3.7.0: pytest is failing in `tests/test_pytest_mock.py::test_standalone_mock` unit

I'm trying to package your module as an rpm package. So I'm using the typical PEP517 based build, install and test cycle used on building packages from non-root account.
- `python3 -sBm build -w --no-isolation`
- because I'm calling `build` with `--no-isolation` I'm using during all processes only locally installed modules
- install .whl file in </install/prefix>
- run pytest with PYTHONPATH pointing to sitearch and sitelib inside </install/prefix>

Here is pytest output:
```console
+ PYTHONPATH=/home/tkloczko/rpmbuild/BUILDROOT/python-pytest-mock-3.7.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages:/home/tkloczko/rpmbuild/BUILDROOT/python-pytest-mock-3.7.0-2.fc35.x86_64/usr/lib/python3.8/site-packages
+ /usr/bin/pytest -ra -p no:relaxed
=========================================================================== test session starts ============================================================================
platform linux -- Python 3.8.12, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /home/tkloczko/rpmbuild/BUILD/pytest-mock-3.7.0, configfile: tox.ini
plugins: mock-3.7.0, cov-3.0.0, asyncio-0.17.2
asyncio: mode=auto
collected 72 items

tests/test_pytest_mock.py .........................................................F..............                                                                   [100%]

================================================================================= FAILURES =================================================================================
___________________________________________________________________________ test_standalone_mock ___________________________________________________________________________

testdir = <Testdir local('/tmp/pytest-of-tkloczko/pytest-2/test_standalone_mock0')>

    def test_standalone_mock(testdir: Any) -> None:
        """Check that the "mock_use_standalone" is being used."""
        testdir.makepyfile(
            """
            def test_foo(mocker):
                pass
        """
        )
        testdir.makeini(
            """
            [pytest]
            mock_use_standalone_module = true
        """
        )
        result = testdir.runpytest_subprocess()
>       assert result.ret == 3
E       assert <ExitCode.OK: 0> == 3
E        +  where <ExitCode.OK: 0> = <RunResult ret=ExitCode.OK len(stdout.lines)=20 len(stderr.lines)=0 duration=0.46s>.ret

/home/tkloczko/rpmbuild/BUILD/pytest-mock-3.7.0/tests/test_pytest_mock.py:723: AssertionError
--------------------------------------------------------------------------- Captured stdout call ---------------------------------------------------------------------------
running: /usr/bin/python3 -mpytest --basetemp=/tmp/pytest-of-tkloczko/pytest-2/test_standalone_mock0/runpytest-0
     in: /tmp/pytest-of-tkloczko/pytest-2/test_standalone_mock0
============================= test session starts ==============================
platform linux -- Python 3.8.12, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /tmp/pytest-of-tkloczko/pytest-2/test_standalone_mock0, configfile: tox.ini
plugins: mock-3.7.0, cov-3.0.0, asyncio-0.17.2
asyncio: mode=legacy
collected 1 item

test_standalone_mock.py .                                                [100%]

=============================== warnings summary ===============================
../../../../usr/lib/python3.8/site-packages/pytest_asyncio/plugin.py:191
  /usr/lib/python3.8/site-packages/pytest_asyncio/plugin.py:191: DeprecationWarning: The 'asyncio_mode' default value will change to 'strict' in future, please explicitly use 'asyncio_mode=strict' or 'asyncio_mode=auto' in pytest configuration file.
    config.issue_config_time_warning(LEGACY_MODE, stacklevel=2)

test_standalone_mock.py::test_foo
  /usr/lib/python3.8/site-packages/pytest_asyncio/plugin.py:317: DeprecationWarning: '@pytest.fixture' is applied to <fixture _mocker, file=/home/tkloczko/rpmbuild/BUILDROOT/python-pytest-mock-3.7.0-2.fc35.x86_64/usr/lib/python3.8/site-packages/pytest_mock/plugin.py, line=388> in 'legacy' mode, please replace it with '@pytest_asyncio.fixture' as a preparation for switching to 'strict' mode (or use 'auto' mode to seamlessly handle all these fixtures as asyncio-driven).
    warnings.warn(

-- Docs: https://docs.pytest.org/en/stable/warnings.html
======================== 1 passed, 2 warnings in 0.03s =========================
========================================================================= short test summary info ==========================================================================
FAILED tests/test_pytest_mock.py::test_standalone_mock - assert <ExitCode.OK: 0> == 3
======================================================================= 1 failed, 71 passed in 6.16s =======================================================================
/usr/lib/python3.8/site-packages/_pytest/pathlib.py:80: PytestWarning: (rm_rf) error removing /tmp/pytest-of-tkloczko/garbage-6f5bceaa-27c1-4828-a3c4-0d1d4c62f501
<class 'OSError'>: [Errno 39] Directory not empty: '/tmp/pytest-of-tkloczko/garbage-6f5bceaa-27c1-4828-a3c4-0d1d4c62f501'
  warnings.warn(
/usr/lib/python3.8/site-packages/_pytest/pathlib.py:80: PytestWarning: (rm_rf) error removing /tmp/pytest-of-tkloczko/garbage-f0b2eb80-1df2-41e3-b9c6-7ae4bdb46198
<class 'OSError'>: [Errno 39] Directory not empty: '/tmp/pytest-of-tkloczko/garbage-f0b2eb80-1df2-41e3-b9c6-7ae4bdb46198'
  warnings.warn(
/usr/lib/python3.8/site-packages/_pytest/pathlib.py:80: PytestWarning: (rm_rf) error removing /tmp/pytest-of-tkloczko/garbage-aa20aba2-7e0e-4ebb-8719-a5e007e0d1c4
<class 'OSError'>: [Errno 39] Directory not empty: '/tmp/pytest-of-tkloczko/garbage-aa20aba2-7e0e-4ebb-8719-a5e007e0d1c4'
  warnings.warn(
/usr/lib/python3.8/site-packages/_pytest/pathlib.py:80: PytestWarning: (rm_rf) error removing /tmp/pytest-of-tkloczko/garbage-d429fee2-5f3d-428e-92a2-6e0fc3e29972
<class 'OSError'>: [Errno 39] Directory not empty: '/tmp/pytest-of-tkloczko/garbage-d429fee2-5f3d-428e-92a2-6e0fc3e29972'
  warnings.warn(
/usr/lib/python3.8/site-packages/_pytest/pathlib.py:80: PytestWarning: (rm_rf) error removing /tmp/pytest-of-tkloczko/garbage-0e7d96b3-02dc-4fec-88d8-b31f111ddb5d
<class 'OSError'>: [Errno 39] Directory not empty: '/tmp/pytest-of-tkloczko/garbage-0e7d96b3-02dc-4fec-88d8-b31f111ddb5d'
  warnings.warn(
/usr/lib/python3.8/site-packages/_pytest/pathlib.py:80: PytestWarning: (rm_rf) error removing /tmp/pytest-of-tkloczko/garbage-9f197e27-3fb6-4540-9abc-37cc28021b4e
<class 'OSError'>: [Errno 39] Directory not empty: '/tmp/pytest-of-tkloczko/garbage-9f197e27-3fb6-4540-9abc-37cc28021b4e'
  warnings.warn(
```

Even with installed 'pytest-mock` pytest is failing.
Additionally please have look on pytest warning.
```console
+ PYTHONPATH=/home/tkloczko/rpmbuild/BUILDROOT/python-pytest-mock-3.7.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages:/home/tkloczko/rpmbuild/BUILDROOT/python-pytest-mock-3.7.0-2.fc35.x86_64/usr/lib/python3.8/site-packages
+ /usr/bin/pytest -ra -p no:relaxed --deselect tests/test_pytest_mock.py::test_standalone_mock
=========================================================================== test session starts ============================================================================
platform linux -- Python 3.8.12, pytest-7.1.1, pluggy-1.0.0
rootdir: /home/tkloczko/rpmbuild/BUILD/pytest-mock-3.7.0, configfile: tox.ini
plugins: mock-3.7.0, asyncio-0.18.2
asyncio: mode=auto
collected 72 items / 1 deselected / 71 selected

tests/test_pytest_mock.py .........................................................FF............                                                                    [100%]

================================================================================= FAILURES =================================================================================
_______________________________________________________________________ test_detailed_introspection ________________________________________________________________________

testdir = <Testdir local('/tmp/pytest-of-tkloczko/pytest-154/test_detailed_introspection0')>

    @pytest.mark.usefixtures("needs_assert_rewrite")
    def test_detailed_introspection(testdir: Any) -> None:
        """Check that the "mock_use_standalone" is being used."""
        testdir.makeini(
            """
            [pytest]
            asyncio_mode=auto
            """
        )
        testdir.makepyfile(
            """
            def test(mocker):
                m = mocker.Mock()
                m('fo')
                m.assert_called_once_with('', bar=4)
        """
        )
        result = testdir.runpytest("-s")
        if NEW_FORMATTING:
            expected_lines = [
                "*AssertionError: expected call not found.",
                "*Expected: mock('', bar=4)",
                "*Actual: mock('fo')",
            ]
        else:
            expected_lines = [
                "*AssertionError: Expected call: mock('', bar=4)*",
                "*Actual call: mock('fo')*",
            ]
        expected_lines += [
            "*pytest introspection follows:*",
            "*Args:",
            "*assert ('fo',) == ('',)",
            "*At index 0 diff: 'fo' != ''*",
            "*Use -v to get the full diff*",
            "*Kwargs:*",
            "*assert {} == {'bar': 4}*",
            "*Right contains* more item*",
            "*{'bar': 4}*",
            "*Use -v to get the full diff*",
        ]
>       result.stdout.fnmatch_lines(expected_lines)
E       Failed: nomatch: '*AssertionError: expected call not found.'
E           and: '============================= test session starts =============================='
E           and: 'platform linux -- Python 3.8.12, pytest-7.1.1, pluggy-1.0.0'
E           and: 'rootdir: /tmp/pytest-of-tkloczko/pytest-154/test_detailed_introspection0, configfile: tox.ini'
E           and: 'plugins: mock-3.7.0, asyncio-0.18.2'
E           and: 'asyncio: mode=auto'
E           and: 'collected 1 item'
E           and: ''
E           and: 'test_detailed_introspection.py F'
E           and: ''
E           and: '=================================== FAILURES ==================================='
E           and: '_____________________________________ test _____________________________________'
E           and: ''
E           and: '__wrapped_mock_method__ = <function NonCallableMock.assert_called_with at 0x7f9ad697b5e0>'
E           and: "args = (<Mock id='140303000159664'>, ''), kwargs = {'bar': 4}"
E           and: '__tracebackhide__ = True'
E           and: 'msg = "expected call not found.\\nExpected: mock(\'\', bar=4)\\nActual: mock(\'fo\')\\n\\npytest introspection follows:\\n\\nArgs:\\nas...et more diff\\nKwargs:\\nassert {} == {\'bar\': 4}\\n  Right contains 1 more item:\\n  {\'bar\': 4}\\n  Use -v to get more diff"'
E           and: "__mock_self = <Mock id='140303000159664'>, actual_args = ('fo',)"
E           and: 'actual_kwargs = {}'
E           and: 'introspection = "\\nArgs:\\nassert (\'fo\',) == (\'\',)\\n  At index 0 diff: \'fo\' != \'\'\\n  Use -v to get more diff\\nKwargs:\\nassert {} == {\'bar\': 4}\\n  Right contains 1 more item:\\n  {\'bar\': 4}\\n  Use -v to get more diff"'
E           and: "@py_assert2 = ('',), @py_assert1 = False"
E           and: '@py_format4 = "(\'fo\',) == (\'\',)\\n~At index 0 diff: \'fo\' != \'\'\\n~Use -v to get more diff"'
E           and: ''
E           and: '    def assert_wrapper('
E           and: '        __wrapped_mock_method__: Callable[..., Any], *args: Any, **kwargs: Any'
E           and: '    ) -> None:'
E           and: '        __tracebackhide__ = True'
E           and: '        try:'
E           and: '>           __wrapped_mock_method__(*args, **kwargs)'
E           and: ''
E           and: '/home/tkloczko/rpmbuild/BUILDROOT/python-pytest-mock-3.7.0-2.fc35.x86_64/usr/lib/python3.8/site-packages/pytest_mock/plugin.py:414: '
E           and: '_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ '
E           and: ''
E           and: "self = <Mock id='140303000159664'>, args = ('',), kwargs = {'bar': 4}"
E           and: "expected = (('',), {'bar': 4}), actual = call('fo')"
E           and: '_error_message = <function NonCallableMock.assert_called_with.<locals>._error_message at 0x7f9ad67d1670>'
E           and: 'cause = None'
E           and: ''
E           and: '    def assert_called_with(self, /, *args, **kwargs):'
E           and: '        """assert that the last call was made with the specified arguments.'
E           and: '    '
E           and: '        Raises an AssertionError if the args and keyword args passed in are'
E           and: '        different to the last call to the mock."""'
E           and: '        if self.call_args is None:'
E           and: '            expected = self._format_mock_call_signature(args, kwargs)'
E           and: "            actual = 'not called.'"
E           and: "            error_message = ('expected call not found.\\nExpected: %s\\nActual: %s'"
E           and: '                    % (expected, actual))'
E           and: '            raise AssertionError(error_message)'
E           and: '    '
E           and: '        def _error_message():'
E           and: '            msg = self._format_mock_failure_message(args, kwargs)'
E           and: '            return msg'
E           and: '        expected = self._call_matcher((args, kwargs))'
E           and: '        actual = self._call_matcher(self.call_args)'
E           and: '        if expected != actual:'
E           and: '            cause = expected if isinstance(expected, Exception) else None'
E           and: '>           raise AssertionError(_error_message()) from cause'
E       fnmatch: '*AssertionError: expected call not found.'
E          with: 'E           AssertionError: expected call not found.'
E       fnmatch: "*Expected: mock('', bar=4)"
E          with: "E           Expected: mock('', bar=4)"
E       fnmatch: "*Actual: mock('fo')"
E          with: "E           Actual: mock('fo')"
E       nomatch: '*pytest introspection follows:*'
E           and: ''
E           and: '/usr/lib64/python3.8/unittest/mock.py:913: AssertionError'
E           and: ''
E           and: 'During handling of the above exception, another exception occurred:'
E           and: ''
E           and: '__wrapped_mock_method__ = <function NonCallableMock.assert_called_once_with at 0x7f9ad697b670>'
E           and: "args = (<Mock id='140303000159664'>, ''), kwargs = {'bar': 4}"
E           and: '__tracebackhide__ = True'
E       fnmatch: '*pytest introspection follows:*'
E          with: 'msg = "expected call not found.\\nExpected: mock(\'\', bar=4)\\nActual: mock(\'fo\')\\n\\npytest introspection follows:\\n\\nArgs:\\nas...et more diff\\nKwargs:\\nassert {} == {\'bar\': 4}\\n  Right contains 1 more item:\\n  {\'bar\': 4}\\n  Use -v to get more diff"'
E       nomatch: '*Args:'
E           and: ''
E           and: '    def assert_wrapper('
E           and: '        __wrapped_mock_method__: Callable[..., Any], *args: Any, **kwargs: Any'
E           and: '    ) -> None:'
E           and: '        __tracebackhide__ = True'
E           and: '        try:'
E           and: '>           __wrapped_mock_method__(*args, **kwargs)'
E           and: ''
E           and: '/home/tkloczko/rpmbuild/BUILDROOT/python-pytest-mock-3.7.0-2.fc35.x86_64/usr/lib/python3.8/site-packages/pytest_mock/plugin.py:414: '
E           and: '_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ '
E           and: ''
E           and: "self = <Mock id='140303000159664'>, args = ('',), kwargs = {'bar': 4}"
E           and: ''
E           and: '    def assert_called_once_with(self, /, *args, **kwargs):'
E           and: '        """assert that the mock was called exactly once and that that call was'
E           and: '        with the specified arguments."""'
E           and: '        if not self.call_count == 1:'
E           and: '            msg = ("Expected \'%s\' to be called once. Called %s times.%s"'
E           and: "                   % (self._mock_name or 'mock',"
E           and: '                      self.call_count,'
E           and: '                      self._calls_repr()))'
E           and: '            raise AssertionError(msg)'
E           and: '>       return self.assert_called_with(*args, **kwargs)'
E           and: ''
E           and: '/usr/lib64/python3.8/unittest/mock.py:925: '
E           and: '_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ '
E           and: ''
E           and: "args = (<Mock id='140303000159664'>, ''), kwargs = {'bar': 4}"
E           and: '__tracebackhide__ = True'
E           and: ''
E           and: '    def wrap_assert_called_with(*args: Any, **kwargs: Any) -> None:'
E           and: '        __tracebackhide__ = True'
E           and: '>       assert_wrapper(_mock_module_originals["assert_called_with"], *args, **kwargs)'
E           and: ''
E           and: '/home/tkloczko/rpmbuild/BUILDROOT/python-pytest-mock-3.7.0-2.fc35.x86_64/usr/lib/python3.8/site-packages/pytest_mock/plugin.py:447: '
E           and: '_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ '
E           and: ''
E           and: '__wrapped_mock_method__ = <function NonCallableMock.assert_called_with at 0x7f9ad697b5e0>'
E           and: "args = (<Mock id='140303000159664'>, ''), kwargs = {'bar': 4}"
E           and: '__tracebackhide__ = True'
E           and: 'msg = "expected call not found.\\nExpected: mock(\'\', bar=4)\\nActual: mock(\'fo\')\\n\\npytest introspection follows:\\n\\nArgs:\\nas...et more diff\\nKwargs:\\nassert {} == {\'bar\': 4}\\n  Right contains 1 more item:\\n  {\'bar\': 4}\\n  Use -v to get more diff"'
E           and: "__mock_self = <Mock id='140303000159664'>, actual_args = ('fo',)"
E           and: 'actual_kwargs = {}'
E           and: 'introspection = "\\nArgs:\\nassert (\'fo\',) == (\'\',)\\n  At index 0 diff: \'fo\' != \'\'\\n  Use -v to get more diff\\nKwargs:\\nassert {} == {\'bar\': 4}\\n  Right contains 1 more item:\\n  {\'bar\': 4}\\n  Use -v to get more diff"'
E           and: "@py_assert2 = ('',), @py_assert1 = False"
E           and: '@py_format4 = "(\'fo\',) == (\'\',)\\n~At index 0 diff: \'fo\' != \'\'\\n~Use -v to get more diff"'
E           and: ''
E           and: '    def assert_wrapper('
E           and: '        __wrapped_mock_method__: Callable[..., Any], *args: Any, **kwargs: Any'
E           and: '    ) -> None:'
E           and: '        __tracebackhide__ = True'
E           and: '        try:'
E           and: '            __wrapped_mock_method__(*args, **kwargs)'
E           and: '            return'
E           and: '        except AssertionError as e:'
E           and: '            if getattr(e, "_mock_introspection_applied", 0):'
E           and: '                msg = str(e)'
E           and: '            else:'
E           and: '                __mock_self = args[0]'
E           and: '                msg = str(e)'
E           and: '                if __mock_self.call_args is not None:'
E           and: '                    actual_args, actual_kwargs = __mock_self.call_args'
E           and: '                    introspection = ""'
E           and: '                    try:'
E           and: '                        assert actual_args == args[1:]'
E           and: '                    except AssertionError as e_args:'
E           and: '                        introspection += "\\nArgs:\\n" + str(e_args)'
E           and: '                    try:'
E           and: '                        assert actual_kwargs == kwargs'
E           and: '                    except AssertionError as e_kwargs:'
E           and: '                        introspection += "\\nKwargs:\\n" + str(e_kwargs)'
E           and: '                    if introspection:'
E           and: '                        msg += "\\n\\npytest introspection follows:\\n" + introspection'
E           and: '            e = AssertionError(msg)'
E           and: '            e._mock_introspection_applied = True  # type:ignore[attr-defined]'
E           and: '>           raise e'
E           and: 'E           AssertionError: expected call not found.'
E           and: "E           Expected: mock('', bar=4)"
E           and: "E           Actual: mock('fo')"
E           and: 'E           '
E           and: 'E           pytest introspection follows:'
E           and: 'E           '
E       fnmatch: '*Args:'
E          with: 'E           Args:'
E       fnmatch: "*assert ('fo',) == ('',)"
E          with: "E           assert ('fo',) == ('',)"
E       fnmatch: "*At index 0 diff: 'fo' != ''*"
E          with: "E             At index 0 diff: 'fo' != ''"
E       nomatch: '*Use -v to get the full diff*'
E           and: 'E             Use -v to get more diff'
E           and: 'E           Kwargs:'
E           and: "E           assert {} == {'bar': 4}"
E           and: 'E             Right contains 1 more item:'
E           and: "E             {'bar': 4}"
E           and: 'E             Use -v to get more diff'
E           and: ''
E           and: '/home/tkloczko/rpmbuild/BUILDROOT/python-pytest-mock-3.7.0-2.fc35.x86_64/usr/lib/python3.8/site-packages/pytest_mock/plugin.py:437: AssertionError'
E           and: ''
E           and: 'During handling of the above exception, another exception occurred:'
E           and: ''
E           and: 'mocker = <pytest_mock.plugin.MockerFixture object at 0x7f9ad67c4d90>'
E           and: ''
E           and: '    def test(mocker):'
E           and: '        m = mocker.Mock()'
E           and: "        m('fo')"
E           and: ">       m.assert_called_once_with('', bar=4)"
E           and: 'E       AssertionError: expected call not found.'
E           and: "E       Expected: mock('', bar=4)"
E           and: "E       Actual: mock('fo')"
E           and: 'E       '
E           and: 'E       pytest introspection follows:'
E           and: 'E       '
E           and: 'E       Args:'
E           and: "E       assert ('fo',) == ('',)"
E           and: "E         At index 0 diff: 'fo' != ''"
E           and: 'E         Use -v to get more diff'
E           and: 'E       Kwargs:'
E           and: "E       assert {} == {'bar': 4}"
E           and: 'E         Right contains 1 more item:'
E           and: "E         {'bar': 4}"
E           and: 'E         Use -v to get more diff'
E           and: ''
E           and: 'test_detailed_introspection.py:4: AssertionError'
E           and: '=========================== short test summary info ============================'
E           and: 'FAILED test_detailed_introspection.py::test - AssertionError: expected call n...'
E           and: '============================== 1 failed in 0.17s ==============================='
E       remains unmatched: '*Use -v to get the full diff*'

/home/tkloczko/rpmbuild/BUILD/pytest-mock-3.7.0/tests/test_pytest_mock.py:768: Failed
--------------------------------------------------------------------------- Captured stdout call ---------------------------------------------------------------------------
============================= test session starts ==============================
platform linux -- Python 3.8.12, pytest-7.1.1, pluggy-1.0.0
rootdir: /tmp/pytest-of-tkloczko/pytest-154/test_detailed_introspection0, configfile: tox.ini
plugins: mock-3.7.0, asyncio-0.18.2
asyncio: mode=auto
collected 1 item

test_detailed_introspection.py F

=================================== FAILURES ===================================
_____________________________________ test _____________________________________

__wrapped_mock_method__ = <function NonCallableMock.assert_called_with at 0x7f9ad697b5e0>
args = (<Mock id='140303000159664'>, ''), kwargs = {'bar': 4}
__tracebackhide__ = True
msg = "expected call not found.\nExpected: mock('', bar=4)\nActual: mock('fo')\n\npytest introspection follows:\n\nArgs:\nas...et more diff\nKwargs:\nassert {} == {'bar': 4}\n  Right contains 1 more item:\n  {'bar': 4}\n  Use -v to get more diff"
__mock_self = <Mock id='140303000159664'>, actual_args = ('fo',)
actual_kwargs = {}
introspection = "\nArgs:\nassert ('fo',) == ('',)\n  At index 0 diff: 'fo' != ''\n  Use -v to get more diff\nKwargs:\nassert {} == {'bar': 4}\n  Right contains 1 more item:\n  {'bar': 4}\n  Use -v to get more diff"
@py_assert2 = ('',), @py_assert1 = False
@py_format4 = "('fo',) == ('',)\n~At index 0 diff: 'fo' != ''\n~Use -v to get more diff"

    def assert_wrapper(
        __wrapped_mock_method__: Callable[..., Any], *args: Any, **kwargs: Any
    ) -> None:
        __tracebackhide__ = True
        try:
>           __wrapped_mock_method__(*args, **kwargs)

/home/tkloczko/rpmbuild/BUILDROOT/python-pytest-mock-3.7.0-2.fc35.x86_64/usr/lib/python3.8/site-packages/pytest_mock/plugin.py:414:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <Mock id='140303000159664'>, args = ('',), kwargs = {'bar': 4}
expected = (('',), {'bar': 4}), actual = call('fo')
_error_message = <function NonCallableMock.assert_called_with.<locals>._error_message at 0x7f9ad67d1670>
cause = None

    def assert_called_with(self, /, *args, **kwargs):
        """assert that the last call was made with the specified arguments.

        Raises an AssertionError if the args and keyword args passed in are
        different to the last call to the mock."""
        if self.call_args is None:
            expected = self._format_mock_call_signature(args, kwargs)
            actual = 'not called.'
            error_message = ('expected call not found.\nExpected: %s\nActual: %s'
                    % (expected, actual))
            raise AssertionError(error_message)

        def _error_message():
            msg = self._format_mock_failure_message(args, kwargs)
            return msg
        expected = self._call_matcher((args, kwargs))
        actual = self._call_matcher(self.call_args)
        if expected != actual:
            cause = expected if isinstance(expected, Exception) else None
>           raise AssertionError(_error_message()) from cause
E           AssertionError: expected call not found.
E           Expected: mock('', bar=4)
E           Actual: mock('fo')

/usr/lib64/python3.8/unittest/mock.py:913: AssertionError

During handling of the above exception, another exception occurred:

__wrapped_mock_method__ = <function NonCallableMock.assert_called_once_with at 0x7f9ad697b670>
args = (<Mock id='140303000159664'>, ''), kwargs = {'bar': 4}
__tracebackhide__ = True
msg = "expected call not found.\nExpected: mock('', bar=4)\nActual: mock('fo')\n\npytest introspection follows:\n\nArgs:\nas...et more diff\nKwargs:\nassert {} == {'bar': 4}\n  Right contains 1 more item:\n  {'bar': 4}\n  Use -v to get more diff"

    def assert_wrapper(
        __wrapped_mock_method__: Callable[..., Any], *args: Any, **kwargs: Any
    ) -> None:
        __tracebackhide__ = True
        try:
>           __wrapped_mock_method__(*args, **kwargs)

/home/tkloczko/rpmbuild/BUILDROOT/python-pytest-mock-3.7.0-2.fc35.x86_64/usr/lib/python3.8/site-packages/pytest_mock/plugin.py:414:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <Mock id='140303000159664'>, args = ('',), kwargs = {'bar': 4}

    def assert_called_once_with(self, /, *args, **kwargs):
        """assert that the mock was called exactly once and that that call was
        with the specified arguments."""
        if not self.call_count == 1:
            msg = ("Expected '%s' to be called once. Called %s times.%s"
                   % (self._mock_name or 'mock',
                      self.call_count,
                      self._calls_repr()))
            raise AssertionError(msg)
>       return self.assert_called_with(*args, **kwargs)

/usr/lib64/python3.8/unittest/mock.py:925:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

args = (<Mock id='140303000159664'>, ''), kwargs = {'bar': 4}
__tracebackhide__ = True

    def wrap_assert_called_with(*args: Any, **kwargs: Any) -> None:
        __tracebackhide__ = True
>       assert_wrapper(_mock_module_originals["assert_called_with"], *args, **kwargs)

/home/tkloczko/rpmbuild/BUILDROOT/python-pytest-mock-3.7.0-2.fc35.x86_64/usr/lib/python3.8/site-packages/pytest_mock/plugin.py:447:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

__wrapped_mock_method__ = <function NonCallableMock.assert_called_with at 0x7f9ad697b5e0>
args = (<Mock id='140303000159664'>, ''), kwargs = {'bar': 4}
__tracebackhide__ = True
msg = "expected call not found.\nExpected: mock('', bar=4)\nActual: mock('fo')\n\npytest introspection follows:\n\nArgs:\nas...et more diff\nKwargs:\nassert {} == {'bar': 4}\n  Right contains 1 more item:\n  {'bar': 4}\n  Use -v to get more diff"
__mock_self = <Mock id='140303000159664'>, actual_args = ('fo',)
actual_kwargs = {}
introspection = "\nArgs:\nassert ('fo',) == ('',)\n  At index 0 diff: 'fo' != ''\n  Use -v to get more diff\nKwargs:\nassert {} == {'bar': 4}\n  Right contains 1 more item:\n  {'bar': 4}\n  Use -v to get more diff"
@py_assert2 = ('',), @py_assert1 = False
@py_format4 = "('fo',) == ('',)\n~At index 0 diff: 'fo' != ''\n~Use -v to get more diff"

    def assert_wrapper(
        __wrapped_mock_method__: Callable[..., Any], *args: Any, **kwargs: Any
    ) -> None:
        __tracebackhide__ = True
        try:
            __wrapped_mock_method__(*args, **kwargs)
            return
        except AssertionError as e:
            if getattr(e, "_mock_introspection_applied", 0):
                msg = str(e)
            else:
                __mock_self = args[0]
                msg = str(e)
                if __mock_self.call_args is not None:
                    actual_args, actual_kwargs = __mock_self.call_args
                    introspection = ""
                    try:
                        assert actual_args == args[1:]
                    except AssertionError as e_args:
                        introspection += "\nArgs:\n" + str(e_args)
                    try:
                        assert actual_kwargs == kwargs
                    except AssertionError as e_kwargs:
                        introspection += "\nKwargs:\n" + str(e_kwargs)
                    if introspection:
                        msg += "\n\npytest introspection follows:\n" + introspection
            e = AssertionError(msg)
            e._mock_introspection_applied = True  # type:ignore[attr-defined]
>           raise e
E           AssertionError: expected call not found.
E           Expected: mock('', bar=4)
E           Actual: mock('fo')
E
E           pytest introspection follows:
E
E           Args:
E           assert ('fo',) == ('',)
E             At index 0 diff: 'fo' != ''
E             Use -v to get more diff
E           Kwargs:
E           assert {} == {'bar': 4}
E             Right contains 1 more item:
E             {'bar': 4}
E             Use -v to get more diff

/home/tkloczko/rpmbuild/BUILDROOT/python-pytest-mock-3.7.0-2.fc35.x86_64/usr/lib/python3.8/site-packages/pytest_mock/plugin.py:437: AssertionError

During handling of the above exception, another exception occurred:

mocker = <pytest_mock.plugin.MockerFixture object at 0x7f9ad67c4d90>

    def test(mocker):
        m = mocker.Mock()
        m('fo')
>       m.assert_called_once_with('', bar=4)
E       AssertionError: expected call not found.
E       Expected: mock('', bar=4)
E       Actual: mock('fo')
E
E       pytest introspection follows:
E
E       Args:
E       assert ('fo',) == ('',)
E         At index 0 diff: 'fo' != ''
E         Use -v to get more diff
E       Kwargs:
E       assert {} == {'bar': 4}
E         Right contains 1 more item:
E         {'bar': 4}
E         Use -v to get more diff

test_detailed_introspection.py:4: AssertionError
=========================== short test summary info ============================
FAILED test_detailed_introspection.py::test - AssertionError: expected call n...
============================== 1 failed in 0.17s ===============================
____________________________________________________________________ test_detailed_introspection_async _____________________________________________________________________

testdir = <Testdir local('/tmp/pytest-of-tkloczko/pytest-154/test_detailed_introspection_async0')>

    @pytest.mark.skipif(
        sys.version_info < (3, 8), reason="AsyncMock is present on 3.8 and above"
    )
    @pytest.mark.usefixtures("needs_assert_rewrite")
    def test_detailed_introspection_async(testdir: Any) -> None:
        """Check that the "mock_use_standalone" is being used."""
        testdir.makeini(
            """
            [pytest]
            asyncio_mode=auto
            """
        )
        testdir.makepyfile(
            """
            import pytest

            async def test(mocker):
                m = mocker.AsyncMock()
                await m('fo')
                m.assert_awaited_once_with('', bar=4)
        """
        )
        result = testdir.runpytest("-s")
        expected_lines = [
            "*AssertionError: expected await not found.",
            "*Expected: mock('', bar=4)",
            "*Actual: mock('fo')",
            "*pytest introspection follows:*",
            "*Args:",
            "*assert ('fo',) == ('',)",
            "*At index 0 diff: 'fo' != ''*",
            "*Use -v to get the full diff*",
            "*Kwargs:*",
            "*assert {} == {'bar': 4}*",
            "*Right contains* more item*",
            "*{'bar': 4}*",
            "*Use -v to get the full diff*",
        ]
>       result.stdout.fnmatch_lines(expected_lines)
E       Failed: nomatch: '*AssertionError: expected await not found.'
E           and: '============================= test session starts =============================='
E           and: 'platform linux -- Python 3.8.12, pytest-7.1.1, pluggy-1.0.0'
E           and: 'rootdir: /tmp/pytest-of-tkloczko/pytest-154/test_detailed_introspection_async0, configfile: tox.ini'
E           and: 'plugins: mock-3.7.0, asyncio-0.18.2'
E           and: 'asyncio: mode=auto'
E           and: 'collected 1 item'
E           and: ''
E           and: 'test_detailed_introspection_async.py F'
E           and: ''
E           and: '=================================== FAILURES ==================================='
E           and: '_____________________________________ test _____________________________________'
E           and: ''
E           and: '__wrapped_mock_method__ = <function AsyncMockMixin.assert_awaited_with at 0x7f9ad6983e50>'
E           and: "args = (<AsyncMock id='140303000592928'>, ''), kwargs = {'bar': 4}"
E           and: '__tracebackhide__ = True'
E           and: 'msg = "expected await not found.\\nExpected: mock(\'\', bar=4)\\nActual: mock(\'fo\')\\n\\npytest introspection follows:\\n\\nArgs:\\na...et more diff\\nKwargs:\\nassert {} == {\'bar\': 4}\\n  Right contains 1 more item:\\n  {\'bar\': 4}\\n  Use -v to get more diff"'
E           and: "__mock_self = <AsyncMock id='140303000592928'>, actual_args = ('fo',)"
E           and: 'actual_kwargs = {}'
E           and: 'introspection = "\\nArgs:\\nassert (\'fo\',) == (\'\',)\\n  At index 0 diff: \'fo\' != \'\'\\n  Use -v to get more diff\\nKwargs:\\nassert {} == {\'bar\': 4}\\n  Right contains 1 more item:\\n  {\'bar\': 4}\\n  Use -v to get more diff"'
E           and: "@py_assert2 = ('',), @py_assert1 = False"
E           and: '@py_format4 = "(\'fo\',) == (\'\',)\\n~At index 0 diff: \'fo\' != \'\'\\n~Use -v to get more diff"'
E           and: ''
E           and: '    def assert_wrapper('
E           and: '        __wrapped_mock_method__: Callable[..., Any], *args: Any, **kwargs: Any'
E           and: '    ) -> None:'
E           and: '        __tracebackhide__ = True'
E           and: '        try:'
E           and: '>           __wrapped_mock_method__(*args, **kwargs)'
E           and: ''
E           and: '/home/tkloczko/rpmbuild/BUILDROOT/python-pytest-mock-3.7.0-2.fc35.x86_64/usr/lib/python3.8/site-packages/pytest_mock/plugin.py:414: '
E           and: '_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ '
E           and: ''
E           and: "self = <AsyncMock id='140303000592928'>, args = ('',), kwargs = {'bar': 4}"
E           and: "expected = (('',), {'bar': 4})"
E           and: '_error_message = <function AsyncMockMixin.assert_awaited_with.<locals>._error_message at 0x7f9ad6854700>'
E           and: "actual = call('fo'), cause = None"
E           and: ''
E           and: '    def assert_awaited_with(self, /, *args, **kwargs):'
E           and: '        """'
E           and: '        Assert that the last await was with the specified arguments.'
E           and: '        """'
E           and: '        if self.await_args is None:'
E           and: '            expected = self._format_mock_call_signature(args, kwargs)'
E           and: "            raise AssertionError(f'Expected await: {expected}\\nNot awaited')"
E           and: '    '
E           and: '        def _error_message():'
E           and: "            msg = self._format_mock_failure_message(args, kwargs, action='await')"
E           and: '            return msg'
E           and: '    '
E           and: '        expected = self._call_matcher((args, kwargs))'
E           and: '        actual = self._call_matcher(self.await_args)'
E           and: '        if expected != actual:'
E           and: '            cause = expected if isinstance(expected, Exception) else None'
E           and: '>           raise AssertionError(_error_message()) from cause'
E       fnmatch: '*AssertionError: expected await not found.'
E          with: 'E           AssertionError: expected await not found.'
E       fnmatch: "*Expected: mock('', bar=4)"
E          with: "E           Expected: mock('', bar=4)"
E       fnmatch: "*Actual: mock('fo')"
E          with: "E           Actual: mock('fo')"
E       nomatch: '*pytest introspection follows:*'
E           and: ''
E           and: '/usr/lib64/python3.8/unittest/mock.py:2186: AssertionError'
E           and: ''
E           and: 'During handling of the above exception, another exception occurred:'
E           and: ''
E           and: '__wrapped_mock_method__ = <function AsyncMockMixin.assert_awaited_once_with at 0x7f9ad6983ee0>'
E           and: "args = (<AsyncMock id='140303000592928'>, ''), kwargs = {'bar': 4}"
E           and: '__tracebackhide__ = True'
E       fnmatch: '*pytest introspection follows:*'
E          with: 'msg = "expected await not found.\\nExpected: mock(\'\', bar=4)\\nActual: mock(\'fo\')\\n\\npytest introspection follows:\\n\\nArgs:\\na...et more diff\\nKwargs:\\nassert {} == {\'bar\': 4}\\n  Right contains 1 more item:\\n  {\'bar\': 4}\\n  Use -v to get more diff"'
E       nomatch: '*Args:'
E           and: ''
E           and: '    def assert_wrapper('
E           and: '        __wrapped_mock_method__: Callable[..., Any], *args: Any, **kwargs: Any'
E           and: '    ) -> None:'
E           and: '        __tracebackhide__ = True'
E           and: '        try:'
E           and: '>           __wrapped_mock_method__(*args, **kwargs)'
E           and: ''
E           and: '/home/tkloczko/rpmbuild/BUILDROOT/python-pytest-mock-3.7.0-2.fc35.x86_64/usr/lib/python3.8/site-packages/pytest_mock/plugin.py:414: '
E           and: '_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ '
E           and: ''
E           and: "self = <AsyncMock id='140303000592928'>, args = ('',), kwargs = {'bar': 4}"
E           and: ''
E           and: '    def assert_awaited_once_with(self, /, *args, **kwargs):'
E           and: '        """'
E           and: '        Assert that the mock was awaited exactly once and with the specified'
E           and: '        arguments.'
E           and: '        """'
E           and: '        if not self.await_count == 1:'
E           and: '            msg = (f"Expected {self._mock_name or \'mock\'} to have been awaited once."'
E           and: '                   f" Awaited {self.await_count} times.")'
E           and: '            raise AssertionError(msg)'
E           and: '>       return self.assert_awaited_with(*args, **kwargs)'
E           and: ''
E           and: '/usr/lib64/python3.8/unittest/mock.py:2197: '
E           and: '_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ '
E           and: ''
E           and: "args = (<AsyncMock id='140303000592928'>, ''), kwargs = {'bar': 4}"
E           and: '__tracebackhide__ = True'
E           and: ''
E           and: '    def wrap_assert_awaited_with(*args: Any, **kwargs: Any) -> None:'
E           and: '        __tracebackhide__ = True'
E           and: '>       assert_wrapper(_mock_module_originals["assert_awaited_with"], *args, **kwargs)'
E           and: ''
E           and: '/home/tkloczko/rpmbuild/BUILDROOT/python-pytest-mock-3.7.0-2.fc35.x86_64/usr/lib/python3.8/site-packages/pytest_mock/plugin.py:482: '
E           and: '_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ '
E           and: ''
E           and: '__wrapped_mock_method__ = <function AsyncMockMixin.assert_awaited_with at 0x7f9ad6983e50>'
E           and: "args = (<AsyncMock id='140303000592928'>, ''), kwargs = {'bar': 4}"
E           and: '__tracebackhide__ = True'
E           and: 'msg = "expected await not found.\\nExpected: mock(\'\', bar=4)\\nActual: mock(\'fo\')\\n\\npytest introspection follows:\\n\\nArgs:\\na...et more diff\\nKwargs:\\nassert {} == {\'bar\': 4}\\n  Right contains 1 more item:\\n  {\'bar\': 4}\\n  Use -v to get more diff"'
E           and: "__mock_self = <AsyncMock id='140303000592928'>, actual_args = ('fo',)"
E           and: 'actual_kwargs = {}'
E           and: 'introspection = "\\nArgs:\\nassert (\'fo\',) == (\'\',)\\n  At index 0 diff: \'fo\' != \'\'\\n  Use -v to get more diff\\nKwargs:\\nassert {} == {\'bar\': 4}\\n  Right contains 1 more item:\\n  {\'bar\': 4}\\n  Use -v to get more diff"'
E           and: "@py_assert2 = ('',), @py_assert1 = False"
E           and: '@py_format4 = "(\'fo\',) == (\'\',)\\n~At index 0 diff: \'fo\' != \'\'\\n~Use -v to get more diff"'
E           and: ''
E           and: '    def assert_wrapper('
E           and: '        __wrapped_mock_method__: Callable[..., Any], *args: Any, **kwargs: Any'
E           and: '    ) -> None:'
E           and: '        __tracebackhide__ = True'
E           and: '        try:'
E           and: '            __wrapped_mock_method__(*args, **kwargs)'
E           and: '            return'
E           and: '        except AssertionError as e:'
E           and: '            if getattr(e, "_mock_introspection_applied", 0):'
E           and: '                msg = str(e)'
E           and: '            else:'
E           and: '                __mock_self = args[0]'
E           and: '                msg = str(e)'
E           and: '                if __mock_self.call_args is not None:'
E           and: '                    actual_args, actual_kwargs = __mock_self.call_args'
E           and: '                    introspection = ""'
E           and: '                    try:'
E           and: '                        assert actual_args == args[1:]'
E           and: '                    except AssertionError as e_args:'
E           and: '                        introspection += "\\nArgs:\\n" + str(e_args)'
E           and: '                    try:'
E           and: '                        assert actual_kwargs == kwargs'
E           and: '                    except AssertionError as e_kwargs:'
E           and: '                        introspection += "\\nKwargs:\\n" + str(e_kwargs)'
E           and: '                    if introspection:'
E           and: '                        msg += "\\n\\npytest introspection follows:\\n" + introspection'
E           and: '            e = AssertionError(msg)'
E           and: '            e._mock_introspection_applied = True  # type:ignore[attr-defined]'
E           and: '>           raise e'
E           and: 'E           AssertionError: expected await not found.'
E           and: "E           Expected: mock('', bar=4)"
E           and: "E           Actual: mock('fo')"
E           and: 'E           '
E           and: 'E           pytest introspection follows:'
E           and: 'E           '
E       fnmatch: '*Args:'
E          with: 'E           Args:'
E       fnmatch: "*assert ('fo',) == ('',)"
E          with: "E           assert ('fo',) == ('',)"
E       fnmatch: "*At index 0 diff: 'fo' != ''*"
E          with: "E             At index 0 diff: 'fo' != ''"
E       nomatch: '*Use -v to get the full diff*'
E           and: 'E             Use -v to get more diff'
E           and: 'E           Kwargs:'
E           and: "E           assert {} == {'bar': 4}"
E           and: 'E             Right contains 1 more item:'
E           and: "E             {'bar': 4}"
E           and: 'E             Use -v to get more diff'
E           and: ''
E           and: '/home/tkloczko/rpmbuild/BUILDROOT/python-pytest-mock-3.7.0-2.fc35.x86_64/usr/lib/python3.8/site-packages/pytest_mock/plugin.py:437: AssertionError'
E           and: ''
E           and: 'During handling of the above exception, another exception occurred:'
E           and: ''
E           and: 'mocker = <pytest_mock.plugin.MockerFixture object at 0x7f9ad682e7f0>'
E           and: ''
E           and: '    async def test(mocker):'
E           and: '        m = mocker.AsyncMock()'
E           and: "        await m('fo')"
E           and: ">       m.assert_awaited_once_with('', bar=4)"
E           and: 'E       AssertionError: expected await not found.'
E           and: "E       Expected: mock('', bar=4)"
E           and: "E       Actual: mock('fo')"
E           and: 'E       '
E           and: 'E       pytest introspection follows:'
E           and: 'E       '
E           and: 'E       Args:'
E           and: "E       assert ('fo',) == ('',)"
E           and: "E         At index 0 diff: 'fo' != ''"
E           and: 'E         Use -v to get more diff'
E           and: 'E       Kwargs:'
E           and: "E       assert {} == {'bar': 4}"
E           and: 'E         Right contains 1 more item:'
E           and: "E         {'bar': 4}"
E           and: 'E         Use -v to get more diff'
E           and: ''
E           and: 'test_detailed_introspection_async.py:6: AssertionError'
E           and: '=========================== short test summary info ============================'
E           and: 'FAILED test_detailed_introspection_async.py::test - AssertionError: expected ...'
E           and: '============================== 1 failed in 0.14s ==============================='
E       remains unmatched: '*Use -v to get the full diff*'

/home/tkloczko/rpmbuild/BUILD/pytest-mock-3.7.0/tests/test_pytest_mock.py:809: Failed
--------------------------------------------------------------------------- Captured stdout call ---------------------------------------------------------------------------
============================= test session starts ==============================
platform linux -- Python 3.8.12, pytest-7.1.1, pluggy-1.0.0
rootdir: /tmp/pytest-of-tkloczko/pytest-154/test_detailed_introspection_async0, configfile: tox.ini
plugins: mock-3.7.0, asyncio-0.18.2
asyncio: mode=auto
collected 1 item

test_detailed_introspection_async.py F

=================================== FAILURES ===================================
_____________________________________ test _____________________________________

__wrapped_mock_method__ = <function AsyncMockMixin.assert_awaited_with at 0x7f9ad6983e50>
args = (<AsyncMock id='140303000592928'>, ''), kwargs = {'bar': 4}
__tracebackhide__ = True
msg = "expected await not found.\nExpected: mock('', bar=4)\nActual: mock('fo')\n\npytest introspection follows:\n\nArgs:\na...et more diff\nKwargs:\nassert {} == {'bar': 4}\n  Right contains 1 more item:\n  {'bar': 4}\n  Use -v to get more diff"
__mock_self = <AsyncMock id='140303000592928'>, actual_args = ('fo',)
actual_kwargs = {}
introspection = "\nArgs:\nassert ('fo',) == ('',)\n  At index 0 diff: 'fo' != ''\n  Use -v to get more diff\nKwargs:\nassert {} == {'bar': 4}\n  Right contains 1 more item:\n  {'bar': 4}\n  Use -v to get more diff"
@py_assert2 = ('',), @py_assert1 = False
@py_format4 = "('fo',) == ('',)\n~At index 0 diff: 'fo' != ''\n~Use -v to get more diff"

    def assert_wrapper(
        __wrapped_mock_method__: Callable[..., Any], *args: Any, **kwargs: Any
    ) -> None:
        __tracebackhide__ = True
        try:
>           __wrapped_mock_method__(*args, **kwargs)

/home/tkloczko/rpmbuild/BUILDROOT/python-pytest-mock-3.7.0-2.fc35.x86_64/usr/lib/python3.8/site-packages/pytest_mock/plugin.py:414:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <AsyncMock id='140303000592928'>, args = ('',), kwargs = {'bar': 4}
expected = (('',), {'bar': 4})
_error_message = <function AsyncMockMixin.assert_awaited_with.<locals>._error_message at 0x7f9ad6854700>
actual = call('fo'), cause = None

    def assert_awaited_with(self, /, *args, **kwargs):
        """
        Assert that the last await was with the specified arguments.
        """
        if self.await_args is None:
            expected = self._format_mock_call_signature(args, kwargs)
            raise AssertionError(f'Expected await: {expected}\nNot awaited')

        def _error_message():
            msg = self._format_mock_failure_message(args, kwargs, action='await')
            return msg

        expected = self._call_matcher((args, kwargs))
        actual = self._call_matcher(self.await_args)
        if expected != actual:
            cause = expected if isinstance(expected, Exception) else None
>           raise AssertionError(_error_message()) from cause
E           AssertionError: expected await not found.
E           Expected: mock('', bar=4)
E           Actual: mock('fo')

/usr/lib64/python3.8/unittest/mock.py:2186: AssertionError

During handling of the above exception, another exception occurred:

__wrapped_mock_method__ = <function AsyncMockMixin.assert_awaited_once_with at 0x7f9ad6983ee0>
args = (<AsyncMock id='140303000592928'>, ''), kwargs = {'bar': 4}
__tracebackhide__ = True
msg = "expected await not found.\nExpected: mock('', bar=4)\nActual: mock('fo')\n\npytest introspection follows:\n\nArgs:\na...et more diff\nKwargs:\nassert {} == {'bar': 4}\n  Right contains 1 more item:\n  {'bar': 4}\n  Use -v to get more diff"

    def assert_wrapper(
        __wrapped_mock_method__: Callable[..., Any], *args: Any, **kwargs: Any
    ) -> None:
        __tracebackhide__ = True
        try:
>           __wrapped_mock_method__(*args, **kwargs)

/home/tkloczko/rpmbuild/BUILDROOT/python-pytest-mock-3.7.0-2.fc35.x86_64/usr/lib/python3.8/site-packages/pytest_mock/plugin.py:414:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <AsyncMock id='140303000592928'>, args = ('',), kwargs = {'bar': 4}

    def assert_awaited_once_with(self, /, *args, **kwargs):
        """
        Assert that the mock was awaited exactly once and with the specified
        arguments.
        """
        if not self.await_count == 1:
            msg = (f"Expected {self._mock_name or 'mock'} to have been awaited once."
                   f" Awaited {self.await_count} times.")
            raise AssertionError(msg)
>       return self.assert_awaited_with(*args, **kwargs)

/usr/lib64/python3.8/unittest/mock.py:2197:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

args = (<AsyncMock id='140303000592928'>, ''), kwargs = {'bar': 4}
__tracebackhide__ = True

    def wrap_assert_awaited_with(*args: Any, **kwargs: Any) -> None:
        __tracebackhide__ = True
>       assert_wrapper(_mock_module_originals["assert_awaited_with"], *args, **kwargs)

/home/tkloczko/rpmbuild/BUILDROOT/python-pytest-mock-3.7.0-2.fc35.x86_64/usr/lib/python3.8/site-packages/pytest_mock/plugin.py:482:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

__wrapped_mock_method__ = <function AsyncMockMixin.assert_awaited_with at 0x7f9ad6983e50>
args = (<AsyncMock id='140303000592928'>, ''), kwargs = {'bar': 4}
__tracebackhide__ = True
msg = "expected await not found.\nExpected: mock('', bar=4)\nActual: mock('fo')\n\npytest introspection follows:\n\nArgs:\na...et more diff\nKwargs:\nassert {} == {'bar': 4}\n  Right contains 1 more item:\n  {'bar': 4}\n  Use -v to get more diff"
__mock_self = <AsyncMock id='140303000592928'>, actual_args = ('fo',)
actual_kwargs = {}
introspection = "\nArgs:\nassert ('fo',) == ('',)\n  At index 0 diff: 'fo' != ''\n  Use -v to get more diff\nKwargs:\nassert {} == {'bar': 4}\n  Right contains 1 more item:\n  {'bar': 4}\n  Use -v to get more diff"
@py_assert2 = ('',), @py_assert1 = False
@py_format4 = "('fo',) == ('',)\n~At index 0 diff: 'fo' != ''\n~Use -v to get more diff"

    def assert_wrapper(
        __wrapped_mock_method__: Callable[..., Any], *args: Any, **kwargs: Any
    ) -> None:
        __tracebackhide__ = True
        try:
            __wrapped_mock_method__(*args, **kwargs)
            return
        except AssertionError as e:
            if getattr(e, "_mock_introspection_applied", 0):
                msg = str(e)
            else:
                __mock_self = args[0]
                msg = str(e)
                if __mock_self.call_args is not None:
                    actual_args, actual_kwargs = __mock_self.call_args
                    introspection = ""
                    try:
                        assert actual_args == args[1:]
                    except AssertionError as e_args:
                        introspection += "\nArgs:\n" + str(e_args)
                    try:
                        assert actual_kwargs == kwargs
                    except AssertionError as e_kwargs:
                        introspection += "\nKwargs:\n" + str(e_kwargs)
                    if introspection:
                        msg += "\n\npytest introspection follows:\n" + introspection
            e = AssertionError(msg)
            e._mock_introspection_applied = True  # type:ignore[attr-defined]
>           raise e
E           AssertionError: expected await not found.
E           Expected: mock('', bar=4)
E           Actual: mock('fo')
E
E           pytest introspection follows:
E
E           Args:
E           assert ('fo',) == ('',)
E             At index 0 diff: 'fo' != ''
E             Use -v to get more diff
E           Kwargs:
E           assert {} == {'bar': 4}
E             Right contains 1 more item:
E             {'bar': 4}
E             Use -v to get more diff

/home/tkloczko/rpmbuild/BUILDROOT/python-pytest-mock-3.7.0-2.fc35.x86_64/usr/lib/python3.8/site-packages/pytest_mock/plugin.py:437: AssertionError

During handling of the above exception, another exception occurred:

mocker = <pytest_mock.plugin.MockerFixture object at 0x7f9ad682e7f0>

    async def test(mocker):
        m = mocker.AsyncMock()
        await m('fo')
>       m.assert_awaited_once_with('', bar=4)
E       AssertionError: expected await not found.
E       Expected: mock('', bar=4)
E       Actual: mock('fo')
E
E       pytest introspection follows:
E
E       Args:
E       assert ('fo',) == ('',)
E         At index 0 diff: 'fo' != ''
E         Use -v to get more diff
E       Kwargs:
E       assert {} == {'bar': 4}
E         Right contains 1 more item:
E         {'bar': 4}
E         Use -v to get more diff

test_detailed_introspection_async.py:6: AssertionError
=========================== short test summary info ============================
FAILED test_detailed_introspection_async.py::test - AssertionError: expected ...
============================== 1 failed in 0.14s ===============================
============================================================================= warnings summary =============================================================================
tests/test_pytest_mock.py::test_patch_context_manager_with_context_manager
  /usr/lib/python3.8/site-packages/_pytest/python.py:192: PytestRemovedIn8Warning: Passing None has been deprecated.
  See https://docs.pytest.org/en/latest/how-to/capture-warnings.html#additional-use-cases-of-warnings-in-tests for alternatives in common use cases.
    result = testfunction(**testargs)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
========================================================================= short test summary info ==========================================================================
FAILED tests/test_pytest_mock.py::test_detailed_introspection - Failed: nomatch: '*AssertionError: expected call not found.'
FAILED tests/test_pytest_mock.py::test_detailed_introspection_async - Failed: nomatch: '*AssertionError: expected await not found.'
========================================================== 2 failed, 69 passed, 1 deselected, 1 warning in 4.32s ===========================================================
```

Even with installed 'mock` pytest is failing.
Additionally please have look on pytest warning.
```console
+ PYTHONPATH=/home/tkloczko/rpmbuild/BUILDROOT/python-pytest-mock-3.7.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages:/home/tkloczko/rpmbuild/BUILDROOT/python-pytest-mock-3.7.0-2.fc35.x86_64/usr/lib/python3.8/site-packages
+ /usr/bin/pytest -ra -p no:relaxed --deselect tests/test_pytest_mock.py::test_detailed_introspection --deselect tests/test_pytest_mock.py::test_detailed_introspection_async
=========================================================================== test session starts ============================================================================
platform linux -- Python 3.8.12, pytest-7.1.1, pluggy-1.0.0
rootdir: /home/tkloczko/rpmbuild/BUILD/pytest-mock-3.7.0, configfile: tox.ini
plugins: mock-3.7.0, asyncio-0.18.2
asyncio: mode=auto
collected 72 items / 2 deselected / 70 selected

tests/test_pytest_mock.py .........................................................F............                                                                     [100%]

================================================================================= FAILURES =================================================================================
___________________________________________________________________________ test_standalone_mock ___________________________________________________________________________

testdir = <Testdir local('/tmp/pytest-of-tkloczko/pytest-155/test_standalone_mock0')>

    def test_standalone_mock(testdir: Any) -> None:
        """Check that the "mock_use_standalone" is being used."""
        testdir.makepyfile(
            """
            def test_foo(mocker):
                pass
        """
        )
        testdir.makeini(
            """
            [pytest]
            mock_use_standalone_module = true
        """
        )
        result = testdir.runpytest_subprocess()
>       assert result.ret == 3
E       assert <ExitCode.OK: 0> == 3
E        +  where <ExitCode.OK: 0> = <RunResult ret=ExitCode.OK len(stdout.lines)=16 len(stderr.lines)=0 duration=0.37s>.ret

/home/tkloczko/rpmbuild/BUILD/pytest-mock-3.7.0/tests/test_pytest_mock.py:723: AssertionError
--------------------------------------------------------------------------- Captured stdout call ---------------------------------------------------------------------------
running: /usr/bin/python3 -mpytest --basetemp=/tmp/pytest-of-tkloczko/pytest-155/test_standalone_mock0/runpytest-0
     in: /tmp/pytest-of-tkloczko/pytest-155/test_standalone_mock0
============================= test session starts ==============================
platform linux -- Python 3.8.12, pytest-7.1.1, pluggy-1.0.0
rootdir: /tmp/pytest-of-tkloczko/pytest-155/test_standalone_mock0, configfile: tox.ini
plugins: mock-3.7.0, asyncio-0.18.2
asyncio: mode=legacy
collected 1 item

test_standalone_mock.py .                                                [100%]

=============================== warnings summary ===============================
../../../../usr/lib/python3.8/site-packages/pytest_asyncio/plugin.py:191
  /usr/lib/python3.8/site-packages/pytest_asyncio/plugin.py:191: DeprecationWarning: The 'asyncio_mode' default value will change to 'strict' in future, please explicitly use 'asyncio_mode=strict' or 'asyncio_mode=auto' in pytest configuration file.
    config.issue_config_time_warning(LEGACY_MODE, stacklevel=2)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
========================= 1 passed, 1 warning in 0.02s =========================
============================================================================= warnings summary =============================================================================
tests/test_pytest_mock.py::test_patch_context_manager_with_context_manager
  /usr/lib/python3.8/site-packages/_pytest/python.py:192: PytestRemovedIn8Warning: Passing None has been deprecated.
  See https://docs.pytest.org/en/latest/how-to/capture-warnings.html#additional-use-cases-of-warnings-in-tests for alternatives in common use cases.
    result = testfunction(**testargs)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
========================================================================= short test summary info ==========================================================================
FAILED tests/test_pytest_mock.py::test_standalone_mock - assert <ExitCode.OK: 0> == 3
========================================================== 1 failed, 69 passed, 2 deselected, 1 warning in 4.36s ===========================================================
```

--- a/tests/test_pytest_mock.py
+++ b/tests/test_pytest_mock.py
@@ -707,10 +707,14 @@ def test_monkeypatch_no_terminal(testdir: Any) -> None:
 
 def test_standalone_mock(testdir: Any) -> None:
     """Check that the "mock_use_standalone" is being used."""
+    pytest.importorskip("mock")
+
     testdir.makepyfile(
         """
+        import mock
+
         def test_foo(mocker):
-            pass
+            assert mock.MagicMock is mocker.MagicMock
     """
     )
     testdir.makeini(
@@ -720,8 +724,7 @@ def test_standalone_mock(testdir: Any) -> None:
     """
     )
     result = testdir.runpytest_subprocess()
-    assert result.ret == 3
-    result.stderr.fnmatch_lines(["*No module named 'mock'*"])
+    assert result.ret == 0
 
 
 @pytest.mark.usefixtures("needs_assert_rewrite")

--- a/tox.ini
+++ b/tox.ini
@@ -6,6 +6,7 @@ envlist = py{37,38,39,310}, norewrite
 passenv = USER USERNAME
 deps =
     coverage
+    mock
     pytest-asyncio
 commands =
     coverage run --append --source={envsitepackagesdir}/pytest_mock -m pytest tests

